pkg/inclusion/paths.go

```diff
@@ -1,17 +1,61 @@
 package inclusion
 
+import (
+	"math"
+
+	"github.com/celestiaorg/celestia-app/pkg/shares"
+)
+
+type path struct {
+	instructions []WalkInstruction
+	row          int
+}
+
+// calculateCommitPaths calculates all of the paths to subtree roots needed to
+// create the commitment for a given message.
+func calculateCommitPaths(origSquareSize, start, msgShareLen int) []path {
+	// todo: make the non-interactive defaults optional. by calculating the
+	// NextAlignedPowerOfTwo, we are forcing use of the non-interactive
+	// defaults. If we want to make this optional in the future, we have to move
+	// this next line out of this function.
+	start, _ = shares.NextAlignedPowerOfTwo(start, msgShareLen, origSquareSize)
+	startRow, endRow := start/origSquareSize, (start+msgShareLen-1)/origSquareSize
+	normalizedStartIndex := start % origSquareSize
+	normalizedEndIndex := (start + msgShareLen) - endRow*origSquareSize
+	paths := []path{}
+	maxDepth := int(math.Log2(float64(origSquareSize)))
+	for i := startRow; i <= endRow; i++ {
+		start, end := 0, origSquareSize
+		if i == startRow {
+			start = normalizedStartIndex
+		}
+		if i == endRow {
+			end = normalizedEndIndex
+		}
+		coord := calculateSubTreeRootCoordinates(maxDepth, start, end)
+		for _, c := range coord {
+			paths = append(paths, path{
+				instructions: genSubTreeRootPath(c.depth, uint(c.position)),
+				row:          i,
+			})
+		}
+	}
+
+	return paths
+}
+
 // genSubTreeRootPath calculates the path to a given subtree root of a node, given the
 // depth and position of the node. note: the root of the tree is depth 0.
 // The following nolint can be removed after this function is used.
 //nolint:unused,deadcode
```

Comment:
[Non Blocking][Suggestion]

Probably can be removed as this function is now called.
```suggestion
```

Reply:
nice! good find, I'll if there is other feedback, I'll include this in this PR, if not, I'll be sure to include this in the next `ProcessProposal` PR as to not dismiss reviews.